system/core/pages.php

```diff
@@ -302,7 +302,9 @@ function newPage($newName) {
 		name - pagename
 	*/
 	function validatePagename($name) {
-		return preg_replace('/\s/', '_', preg_replace('/^\s|\s$|[^\d\w\s\.-_]/i', '', preg_replace('/\s+/', ' ', $name)));
+    $name = preg_replace('/\s+/', ' ', $name); // Collapse multiple spaces
+    $name = preg_replace('/^\s|\s$|[^\s\d\w\-_]/i', '', $name); //remove spaces from start and remove everythin g but alphanumeric _ - and space
```

Comment:
It seems there is an indentation change here.

Reply:
Fixed.